F1WM/Controllers/NewsController.cs

```diff
@@ -49,6 +61,52 @@ public async Task<IActionResult> GetSingle(int id)
 			}
 		}
 
+		[HttpGet("types")]
+		public async Task<IEnumerable<NewsType>> GetTypes()
+		{
+			try
+			{
+				return await service.GetNewsTypes();
+			}
+			catch (Exception ex)
+			{
+				logger.LogError(ex);
+				throw ex;
+			}
+		}
+
+		[HttpGet("tags")]
+		public async Task<IEnumerable<NewsTag>> GetTags([FromQuery(Name = "categoryId")] int? id = null)
```

Comment:
Another thing to consider: paging in this endpoint. For id =null it returns list of 500 entities at once currently.